timestamp-impl/src/main/java/com/palantir/timestamp/RateLimitedTimestampService.java

```diff
@@ -1,264 +0,0 @@
-/*
- * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
- *
- * Licensed under the BSD-3 License (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://opensource.org/licenses/BSD-3-Clause
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package com.palantir.timestamp;
-
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.AtomicReference;
-
-import javax.annotation.Nullable;
-import javax.annotation.concurrent.GuardedBy;
-import javax.annotation.concurrent.ThreadSafe;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import com.google.common.base.Preconditions;
-import com.palantir.common.base.Throwables;
-import com.palantir.common.proxy.TimingProxy;
-import com.palantir.util.jmx.OperationTimer;
-import com.palantir.util.timer.LoggingOperationTimer;
-
-/**
- * This uses smart batching to queue up requests and send them all as one larger batch.
- * @author carrino
- */
-@ThreadSafe
-public class RateLimitedTimestampService implements TimestampService {
```

Comment:
I'd like to request to keep this in the OSS codebase for now, if we're proceeding with this PR in general. We might need/want this optimisation for timelock to be scalable